api/download.py

```diff
@@ -315,3 +315,92 @@ def download(self):
                 log.debug(json.dumps(req_spec, sort_keys=True, indent=4, separators=(',', ': ')))
 
                 return self._preflight_archivestream(req_spec, collection=self.get_param('collection'))
+
+    def summary(self):
+        """Return a summary of what has been/will be downloaded based on a given query"""
+        req = self.request.json_body
+        req['_id'] = bson.ObjectId(req['_id'])
+        level = req['level']
+
+        containers = ['projects', 'sessions', 'acquisitions', 'analyses']
+        cont_query = {}
+        if level == 'projects':
+            # Grab sessions and their ids
+            sessions = config.db.sessions.find({'project': req['_id']}, {'_id': 1})
+            session_ids = [s['_id'] for s in sessions]
+
+            # Grab acquisitions and their ids
+            acquisitions = config.db.acquisitions.find({'session': {'$in': session_ids}}, {'_id': 1})
+            acquisition_ids = [a['_id'] for a in acquisitions]
+            parent_ids = [req['_id']] + session_ids + acquisition_ids
+
+            # # Grab analyses and their ids
+            # analysis_ids = [an['_id'] for an in config.db.analyses.find({'parent.id': {'$in': parent_ids}})]
+
+            # for each type of container below it will have a slightly modified match query
+            cont_query = {
+                'projects': {'_id': req['_id']},
+                'sessions': {'project': req['_id']},
+                'acquisitions': {'session': {'$in': session_ids}},
+                'analyses': {'parent.id': {'$in': parent_ids}}
+            }
+        elif level == 'sessions':
+
+            # Grab acquisitions and their ids
+            acquisitions = config.db.acquisitions.find({'session': req['_id']}, {'_id': 1})
+            acquisition_ids = [a['_id'] for a in acquisitions]
+            parent_ids = [req['_id']] + acquisition_ids
+
+            # # Grab analyses and their ids
```

Comment:
We won't want want analyses included in project/session/acquisition download summary (because they aren't included in the actual download), so it's okay to remove this.